### ISSUE TITLE
Installation error: "babel: not found"

### ISSUE THREAD

**Issue by weijiekoh** [raineorshine/solgraph#27]
```bash
$ sudo yarn global add solgraph

............

[4/4] Building fresh packages...           
[1/5] ⠁ node-sass: node-sass build Binary found at /usr/local/share/.config/yarn/global/node_modules/node-sass/vendor/linux-x64-59/binding.node                             
[-/5] ⠁ waiting...                         
[3/5] ⠁ uws                                
[-/5] ⠁ waiting...                         
error /usr/local/share/.config/yarn/global/node_modules/solgraph: Command failed.     
Exit code: 1                               
Command: npm run build                     
Arguments:                                 
Directory: /usr/local/share/.config/yarn/global/node_modules/solgraph                 
Output:                                    
> solgraph@0.3.0 build /usr/local/share/.config/yarn/global/node_modules/solgraph     
> babel -d dist src                        

sh: 1: babel: not found                    
npm ERR! file sh                           
npm ERR! code ELIFECYCLE                   
npm ERR! errno ENOENT                      
npm ERR! syscall spawn                     
npm ERR! solgraph@0.3.0 build: `babel -d dist src`                                    
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the solgraph@0.3.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I tried to remedy this with `sudo yarn global add babel`, but when I ran `sudo yarn global add solgraph` next, it complained:

```
[4/4] Building fresh packages...
[1/5] ⠈ node-sass: node-sass build Binary found at /usr/local/share/.config/yarn/global/node_modules/node-sass/vendor/linux-x64-59/binding.node
[-/5] ⠈ waiting...
[3/5] ⠈ uws
[-/5] ⠈ waiting...
error /usr/local/share/.config/yarn/global/node_modules/solgraph: Command failed.
Exit code: 1
Command: npm run build
Arguments: 
Directory: /usr/local/share/.config/yarn/global/node_modules/solgraph
Output:
> solgraph@0.3.0 build /usr/local/share/.config/yarn/global/node_modules/solgraph
> babel -d dist src

You have mistakenly installed the `babel` package, which is a no-op in Babel 6.
Babel's CLI commands have been moved from the `babel` package to the `babel-cli` package.

    npm uninstall babel
    npm install --save-dev babel-cli

See http://babeljs.io/docs/usage/cli/ for setup instructions.
```

I then tried:

```
sudo yarn global remove babel && \
sudo yarn global add -D babel && \
sudo yarn global add solgraph
```

But received the initial error again: `sh: 1: babel: not found`

The same issue occurs with `npm`.

**Comment by raineorshine** [comment]
Hmmm. Works for me and CI passes. My babel is located at `./node_modules/.bin/babel`. What happens when you run `type babel`? I'm not sure why it's not getting installed correctly.

@jpantunes Any ideas?

**Comment by jpantunes** [comment]
The package.json file needs to be edited, specifically:
    "build": "babel -d dist src",

Should be:     
    "build": "@babel -d dist src",

It's my mistake, I'll make a pull request in a few minutes.

**Comment by jpantunes** [comment]
It took me a couple of attempts but I was able to reproduce.

The build script expects babel to be available as a global install, and in the latest version the dependencies can be resolved with:

    npm i -g @babel/preset-es2015 @babel/preset-stage-1 @babel/cli @babel/core @babel/register 

I'm not sure if that's the only solution though.

**Comment by raineorshine** [comment]
`./node_modules/.bin` is added to the PATH in npm scripts though, so babel shouldn't have to be global.

**Comment by jpantunes** [comment]
Ok, in that case I believe it's my mistake.  

Installing solgraph 0.3.0 with --only=dev flag works, so I'm convinced the package.json file needs to be updated to include the babel packages as a Dependency because I've listed them under DevDependency only. I'll make a pull request in a few minutes.

**Comment by jpantunes** [comment]
I will need someone else to test as well because I think the behaviour changes depending on whether I use an account with npm-global in the PATH or not.

**Comment by weijiekoh** [comment]
I can test this with a fresh Ubuntu installation in a VM.

How should I go about doing this? I don't think cloning your repo and running `npm install` will reproduce the error, as I've tried that with both jpantunes's `decentralised` branch and raineorshine's `master` branch, without encountering the error.

**Comment by raineorshine** [comment]
#29 merged and published in `v0.3.1`
